publisher-sdk/src/main/java/com/criteo/publisher/Criteo.java

```diff
@@ -74,17 +76,21 @@ public Builder mopubConsent(@Nullable String mopubConsent) {
     }
 
     public Criteo init() throws CriteoInitException {
+      if (adUnits == null) {
+        adUnits = new ArrayList<>();
+      }
       return Criteo.init(application, criteoPublisherId, adUnits, usPrivacyOptOut, mopubConsent);
     }
   }
 
   private static Criteo init(
       @NonNull Application application,
       @NonNull String criteoPublisherId,
-      @Nullable List<AdUnit> adUnits,
+      @NonNull List<AdUnit> adUnits,
       @Nullable Boolean usPrivacyOptOut,
       @Nullable String mopubConsent
   ) throws CriteoInitException {
+    Logger logger = LoggerFactory.getLogger(Criteo.class);
```

Comment:
For consistency with what has been done in the past, why not get the logger through the dependency provider?

Reply:
The logger factory is handled by the dependency provider. `LoggerFactory.getLogger` is a shortcut to `DependencyProvider.getInstance().provideLoggerFactory().createLogger(klass)`.